internal/objects/service/service.go

```diff
@@ -262,6 +260,14 @@ func DesiredEnvoyService(contour *operatorv1alpha1.Contour) *corev1.Service {
 		svc.Spec.Type = corev1.ServiceTypeLoadBalancer
 		provider := contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type
 		lbAnnotations := LbAnnotations[provider]
+		// Add the NLB annotation to the map if specified by AWS provider parameters.
+		if provider == operatorv1alpha1.AWSLoadBalancerProvider &&
+			contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.LoadBalancerServicePublishingType &&
+			contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.AWSLoadBalancerProvider &&
+			contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.AWS != nil &&
```

Comment:
Would this not be obsolete, since next condition already includes `!= nil` condition in it?

Reply:
AWS provider params is an optional field. Checking to make sure it's not nil and the param type is NLB.